tools/shoestring/shoestring/healthagents/rest_https_certificate.py

```diff
@@ -24,25 +26,45 @@ def _openssl_run_sclient_verify(hostname, test_args):
 		'localhost:3001'
 	] + test_args, command_input='Q', show_output=False)
 
+	# verify the certificate is valid
+	for line in lines:
+		line = line.strip()
+		if line.startswith('verify error:'):
+			# normalize error message
+			if '1.1.1' in openssl_executor.version():
+				line = line.replace('self signed', 'self-signed')
+			return False, line
+
+	with tempfile.TemporaryDirectory() as temp_directory:
+		temp_file = Path(temp_directory) / 'cert.txt'
+		with open(temp_file, 'w', encoding='utf-8') as cert_file:
+			cert_file.writelines(lines)
+
+		lines = openssl_executor.dispatch([
+			'x509',
+			'-inform',
+			'pem',
+			'-noout',
+			'-text',
+			'-in',
+			temp_file
+		], command_input='Q', show_output=False)
+
+	searching_for_date = [r'Not Before: (.*)', r'Not After : (.*)']
 	collect_dates = False
 	collected_dates = []
 	for line in lines:
 		line = line.strip()
-		if collect_dates:
-			res = re.search(r'NotBefore: (.*); NotAfter: (.*)', line)
+		if collect_dates and searching_for_date:
+			res = re.search(searching_for_date[0], line)
 			if res:
-				collected_dates.append((parsedate_to_datetime(res.group(1)), parsedate_to_datetime(res.group(2))))
+				collected_dates.append(parsedate_to_datetime(res.group(1)))
+				searching_for_date.pop(0)
 
-		if line == 'Certificate chain':
+		if line == 'Certificate:':
 			collect_dates = True
 
-		if line.startswith('Verify return code:'):
-			if 'Verify return code: 0 (ok)' == line:
-				return True, collected_dates
-
-			return False, line
-
-	return False, 'could not parse s_client response'
+	return (True, collected_dates) if 2 == len(collected_dates) else (False, 'could not parse s_client response')
```

Comment:
original version would collect multiple certificate date ranges

but idk if we need that anymore, so new version is probably fine

cc: @gimre-xymcity

Reply:
I think we do.
IIRC concatenated pem file has CA followed by node cert, and IIRC `x509` only parses FIRST certificate :| (that's why I've used `s_client` output)
There was some other command that could be used, I just don't remember what it was.